src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/data/NotificationRepositoryTest.java

```diff
@@ -55,6 +58,185 @@ void should_return_empty_optional_when_there_is_no_notification_in_db() {
         assertThat(notificationRepository.find(1_000)).isEmpty();
     }
 
+    @Test
+    void should_find_single_notification_by_zip_file_name_and_service() {
+        // given
+        final var zipFileName = "zip_file_name";
+        final var service = "service";
+
+        final var newNotification = new NewNotification(
+            zipFileName,
+            "po_box",
+            service,
+            "dcn",
+            ErrorCode.ERR_AV_FAILED,
+            "error_description"
+        );
+
+        // when
+        notificationRepository.insert(newNotification);
+
+        // and
+        List<Notification> notifications = notificationRepository.find(zipFileName, service);
+
+        // then
+        assertThat(notifications)
+            .isNotEmpty()
+            .hasSize(1)
+            .extracting(this::getTupleFromNotification)
+            .containsExactlyInAnyOrder(
+                tuple(
+                    zipFileName,
+                    newNotification.poBox,
+                    service,
+                    newNotification.documentControlNumber,
+                    newNotification.errorCode,
+                    newNotification.errorDescription,
+                    null,
+                    PENDING
+                )
+            )
+        ;
+    }
+
+    @Test
+    void should_exclude_notifications_from_other_service() {
+        // given
+        final var zipFileName = "zip_file_name";
+        final var service = "service";
+
+        final var newNotification1 = new NewNotification(
+            zipFileName,
+            "po_box1",
+            service,
+            "dcn1",
+            ErrorCode.ERR_AV_FAILED,
+            "error_description1"
+        );
+        final var newNotificationFromOtherService = new NewNotification(
+            zipFileName,
+            "po_box2",
+            "other_service",
+            "dcn2",
+            ErrorCode.ERR_FILE_LIMIT_EXCEEDED,
+            "error_description2"
+        );
+        final var newNotification2 = new NewNotification(
+            zipFileName,
+            "po_box2",
+            service,
+            "dcn2",
+            ErrorCode.ERR_METAFILE_INVALID,
+            "error_description2"
+        );
+
+        // when
+        notificationRepository.insert(newNotification1);
+        notificationRepository.insert(newNotificationFromOtherService);
+        notificationRepository.insert(newNotification2);
+
+        // and
+        List<Notification> notifications = notificationRepository.find(zipFileName, service);
+
+        // then
+        assertThat(notifications)
+            .isNotEmpty()
+            .hasSize(2)
+            .extracting(this::getTupleFromNotification)
+            .containsExactlyInAnyOrder(
+                tuple(
```

Comment:
you could use soft assertions which would tell you everything which is not matching